sql/binlog.cc

```diff
@@ -7169,8 +7300,7 @@ bool MYSQL_BIN_LOG::append_event(Log_event* ev, Master_info *mi)
   DBUG_RETURN(error);
 }
 
-
-bool MYSQL_BIN_LOG::append_buffer(const char* buf, uint len, Master_info *mi)
+bool MYSQL_BIN_LOG::append_buffer(uchar* buf, uint len, Master_info *mi)
```

Comment:
Not only buf= ebuf, also encrypt_event changes buf. This is because encrypt_event moves timestamp in place of event's length - before the buf gets encrypted.